lib/fog/vsphere/requests/compute/list_vm_interfaces.rb

```diff
@@ -61,7 +61,7 @@ def get_raw_interface(vm_id, options = {})
 
         private
 
-        def raw_to_hash(nic, datacenter)
+        def raw_to_hash(nic, datacenter = nil)
```

Comment:
Thanks, is the method called here?

https://github.com/fog/fog-vsphere/blob/74fbbc9241b746c30ee64d2e54cd0310b8b6a6bf/lib/fog/vsphere/requests/compute/list_vm_interfaces.rb#L32

I'm asking because a better fix would be to call the method with passing the datacenter and keeping this as a required parameter.

Reply:
I started with a 100% clone of the master branch on fog-vsphere
Then did a diff with my working patched v3.2.1 and looked at the differences related to the nsx-t patches (files in which I did changes: list_networks.rb and list_vm_interfaces.rb)

The diff between my working v.3.2.1 and the master on fog-vsphere showed only differences in commented code:

`36a37
>           #raw_to_hash(raw) if raw
63a65,66
>         # rubocop:disable Style/ConditionalAssignment
>         #def raw_to_hash(nic)
68a72,74
>           #else
>           #  network = nil
>           #end`

Reply:
Thanks, but I don't quite get what you're trying to tell me.

If we change the line I mentioned to:

```ruby
get_raw_interfaces(vm_id, datacenter).map { |nic| raw_to_hash(nic, datacenter) } 
```

Does that fix the issue on master as well?

Reply:
sorry, was afk for a sec and realized in the meantime my reply was probably unclear, posting diffs for starters are quite silly because github shows diffs already :)
will try the suggested line while reverting the default nil value in the previous change

Reply:
@timogoebel your patch worked :) hope this fix is more optimal